pkg/aws/iam_client.go

```diff
@@ -66,6 +122,24 @@ func (wrapper iamWrapper) CreateUser(userName string) (*types.User, error) {
 	return user, err
 }
 
+func (wrapper iamWrapper) CreateUserIfNotExists(userName string) error {
+	_, err := wrapper.IamClient.CreateUser(context.TODO(), &iam.CreateUserInput{
+		UserName: aws.String(userName),
+	})
+	if err != nil {
+		var apiError smithy.APIError
+		if errors.As(err, &apiError) {
+			switch apiError.(type) {
+			case *types.EntityAlreadyExistsException:
+				return nil
+			default:
+				return err
+			}
+		}
+	}
```

Comment:
I see that this code is repeated in the `CreateLoginProfileIfNotExists` function below. I'd suggest to refactor the code into a function, for example `IsEntityAlreadyExistsException(err error) bool`, so then the code would be more readable, and duplication would be avoided:
```go
if err != nil && !IsEntityAlreadyExistsException(err) {
  return err
}
```